spanner_orm/query.py

```diff
@@ -26,8 +26,15 @@ class SpannerQuery(abc.ABC):
   def __init__(self, model, conditions):
     self._model = model
     self._conditions = conditions
+    self._param_offset = 0
+    self._sql = ''
+    self._parameters = {}
+    self._types = {}
     self._build()
 
+  def _param_index(self):
```

Comment:
name of this method feels confusing. is this supposed to be last parameter's index(+ 1)? so it actually gives the index for where next parameter would go?
the word "index" makes it confusing. it just returns length of parameters with offset in mind. `offsetted_params_length`?

Reply:
the intent here is to return the suffix of the next parameter. maybe _next_param_suffix?

Reply:
That works, then this line:
```
      subquery = _SelectSubQuery(join.destination, join.conditions,
                                  param_offset=self._param_index())
```
would be `param_offset=self._next_param_suffix()`

which becomes confusing because of non-obvious relation between a suffix and an offset. How about `next_param_index`? `next_param_suffix` implies it returns a string, because self.suffix is string, but it returns an int

Reply:
sounds good, i've updated the name